rasterio/_base.pyx

```diff
@@ -809,13 +809,78 @@ cdef class DatasetBase(object):
         num_items = CSLCount(metadata)
         return dict(metadata[i].split('=', 1) for i in range(num_items))
 
-    def colorinterp(self, bidx):
-        """Returns the color interpretation for a band or None."""
-        cdef GDALRasterBandH band = NULL
+    property colorinterp:
 
-        band = self.band(bidx)
-        value = GDALGetRasterColorInterpretation(band)
-        return ColorInterp(value)
+        """Returns a sequence of ``ColorInterp.<enum>`` representing
+        color interpretation in band order.
+
+        Pattern for converting to a band mapping:
+
+            from collections import OrderedDict
+            with rasterio.open('image.tif') as src:
+                band_mapping = OrderedDict(
+                    zip(src.colorinterp, self.indexes))
```

Comment:
@geowurster should this be `src.indexes`?

Reply:
Also looks like it should be `zip(src.indexes, src.colorinterp))`    (doesn't run as written)

Reply:
@brendan-ward Good catch!